File: sponge-nashorn/examples/script/js/plugins_kb.js
/**
 * Sponge Knowledge base
 * Defining plugins in knowledge base.
 */

function onInit() {
    // Variables for assertions only
    sponge.setVariable("valueBefore", null);
    sponge.setVariable("valueAfter", null);
}

// Example plugin defined in the knowledge base.
var ScriptPlugin = Java.extend(Plugin, {
    onConfigure: function(self, configuration) {
        self.logger.debug("configuration {}", self.name);
        self.target = new function() {
            this.storedValue = configuration.getString("storedValue", "default");
            this.getStoredValue = function() {
                return this.storedValue;
            };
            this.setStoredValue = function(value) {
                this.storedValue = value;
            };
        };
    },
    onInit: function(self) {
        self.logger.debug("Initializing {}", self.name);
    },
    onStartup: function(self) {
        self.logger.debug("Starting up {}", self.name);
    }
});

var PluginTrigger = Java.extend(Trigger, {
    onConfigure: function(self) {
        self.withEvent("e1");
    },
    onRun: function(self, event) {
    	valueBefore = scriptPlugin.target.getStoredValue();
    	self.logger.info("Plugin stored value: {}", valueBefore);
    	sponge.setVariable("valueBefore", valueBefore);
    	scriptPlugin.target.setStoredValue(event.get("value"));
    	valueAfter = scriptPlugin.target.getStoredValue();
        self.logger.info("New stored value: {}", valueAfter);
        sponge.setVariable("valueAfter", valueAfter);
    }
});

function onStartup() {
    sponge.event("e1").set("value", "Value B").send();
}

File: sponge-jython/examples/script/py/plugins_kb.py
"""
Sponge Knowledge Base
Defining plugins in knowledge base.
"""

def onInit():
    # Variables for assertions only
    sponge.setVariable("valueBefore", None)
    sponge.setVariable("valueAfter", None)

# Example plugin defined in the knowledge base.
class ScriptPlugin(Plugin):
    def onConfigure(self, configuration):
        self.storedValue = configuration.getString("storedValue", "default")
    def onInit(self):
        self.logger.debug("Initializing {}", self.name)
    def onStartup(self):
        self.logger.debug("Starting up {}", self.name)
    def getStoredValue(self):
        return self.storedValue
    def setStoredValue(self, value):
        self.storedValue = value


class PluginTrigger(Trigger):
    def onConfigure(self):
        self.withEvent("e1")
    def onRun(self, event):
    	valueBefore = scriptPlugin.getStoredValue()
    	self.logger.info("Plugin stored value: {}", valueBefore)
        sponge.setVariable("valueBefore", valueBefore)
    	scriptPlugin.setStoredValue(event.get("value"))
        valueAfter = scriptPlugin.getStoredValue()
        self.logger.info("New stored value: {}", valueAfter)
        sponge.setVariable("valueAfter", valueAfter)

def onStartup():
    sponge.event("e1").set("value", "Value B").send()
